cartridge/failover.lua

```diff
@@ -730,6 +735,22 @@ local function cfg(clusterwide_config)
             end,
         })
         vars.failover_fiber:name('cartridge.stateful-failover')
+    elseif failover_cfg.mode == 'raft' then
+        raft_failover.cfg(failover_cfg)
+
+        vars.fencing_enabled = false
+        vars.consistency_needed = false
+        first_appointments = raft_failover.get_appointments(topology_cfg)
+
+        vars.failover_fiber = fiber.new(failover_loop, {
+            get_appointments = function()
+                vars.membership_notification:wait()
```

Comment:
Do I correctly understand that any notification will be considered as leader change? Why is it not enough to transfer this info via channel from election trigger? Instances in other replicasets have its own leaders that shouldn't be affected if leader of single replicaset change.

Reply:
The current failover mechanism needs to know who is a leader in other replicasets to use this info for rpc calls. Now eventual (and Raft, based on eventual) receives notifications about leader changes and use it to update local leader list.

Reply:
The current failover mechanism needs to know who is a leader in other replicasets to use this info for rpc calls. Now eventual (and Raft, based on eventual) receives notifications about leader changes and use it to update local leader list.

Reply:
Any leader change anyway triggers apply_config. I guess topology should be enough.

Reply:
Suppose there are many instances running on the same host, what happens if the host goes missing?  When for each leader that was on this host, a new one will be elected at the same time, will there not be a race with the loss of actual leader on the other replicasets, or even worse, a  deadloop with a two-phase cfg apply commit?

Everything looks like it need to somehow prevent the massive simultaneous applying the config.

Reply:
> Everything looks like it need to somehow prevent the massive simultaneous applying the config.

It's relevant for any failover type... And probably it's even implemented.

Reply:
As we discussed IRL, leader table not in the topology, so we need membership to notify instances about leader change